src/Altinn.Platform/Altinn.Platform.Register/Register/Program.cs

```diff
@@ -1,164 +1,286 @@
+{
+    logger.LogInformation("Program // ConfigureServices");
+
+    services.AddControllers().AddJsonOptions(options =>
+    {
+        options.JsonSerializerOptions.WriteIndented = true;
+        options.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
+    });
+
+    services.AddMemoryCache();
+    services.AddHealthChecks().AddCheck<HealthCheck>("register_health_check");
+
+    services.AddSingleton(config);
+    services.Configure<GeneralSettings>(config.GetSection("GeneralSettings"));
+    services.Configure<KeyVaultSettings>(config.GetSection("kvSetting"));
+    services.Configure<AccessTokenSettings>(config.GetSection("AccessTokenSettings"));
+    services.Configure<PlatformSettings>(config.GetSection("PlatformSettings"));
+
+    services.AddSingleton<IAuthorizationHandler, AccessTokenHandler>();
+    services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+    services.AddSingleton<ISigningKeysResolver, SigningKeysResolver>();
+
+    services.AddAuthentication(JwtCookieDefaults.AuthenticationScheme)
+          .AddJwtCookie(JwtCookieDefaults.AuthenticationScheme, options =>
+          {
+              GeneralSettings generalSettings = config.GetSection("GeneralSettings").Get<GeneralSettings>();
+              options.JwtCookieName = generalSettings.JwtCookieName;
+              options.MetadataAddress = generalSettings.OpenIdWellKnownEndpoint;
+              options.TokenValidationParameters = new TokenValidationParameters
+              {
+                  ValidateIssuerSigningKey = true,
+                  ValidateIssuer = false,
+                  ValidateAudience = false,
+                  RequireExpirationTime = true,
+                  ValidateLifetime = true,
+                  ClockSkew = TimeSpan.Zero
+              };
+
+              if (builder.Environment.IsDevelopment())
+              {
+                  options.RequireHttpsMetadata = false;
+              }
+          });
+
+    services.AddAuthorization(options =>
+    {
+        options.AddPolicy("PlatformAccess", policy => policy.Requirements.Add(new AccessTokenRequirement()));
+    });
+
+    services.AddHttpClient<IOrganizations, OrganizationsWrapper>();
+    services.AddHttpClient<IParties, PartiesWrapper>();
+    services.AddHttpClient<IAuthorization, AuthorizationWrapper>();
+
+    if (!string.IsNullOrEmpty(applicationInsightsKey))
+    {
+        services.AddSingleton(typeof(ITelemetryChannel), new ServerTelemetryChannel() { StorageFolder = "/tmp/logtelemetry" });
+        services.AddApplicationInsightsTelemetry(applicationInsightsKey);
+        services.AddApplicationInsightsTelemetryProcessor<HealthTelemetryFilter>();
+        services.AddApplicationInsightsTelemetryProcessor<IdentityTelemetryFilter>();
+        services.AddSingleton<ITelemetryInitializer, CustomTelemetryInitializer>();
+
+        logger.LogInformation($"Startup // ApplicationInsightsTelemetryKey = {applicationInsightsKey}");
```

Comment:
We should avoid string interpolation here (when logging). Use the params parameter array instead.